File: cardsSdk/src/darwinMain/kotlin/com/cards/session/cards/network/NetworkConstants.kt
package com.cards.session.cards.network

import platform.Foundation.NSBundle

actual object NetworkConstants {
    actual val PROD_URL: String = "https://api.xendit.co/v3"
    actual val STG_URL: String = "https://api.stg.tidnex.dev/v3"
    actual val BASE_URL: String
        get() = if (isDebugBuild()) STG_URL else PROD_URL

    private fun isDebugBuild(): Boolean {
        val bundle = NSBundle.mainBundle
        return bundle.infoDictionary?.get("Debug") as? Boolean ?: false
    }
} 

File: xendit_cards_session/lib/src/network/network_constants.dart
import 'package:flutter/foundation.dart';

/// Network-related constants
class NetworkConstants {
  /// Base URL for API requests
  /// Returns staging URL in debug mode, production URL otherwise
  static String get baseUrl {
    return kDebugMode 
        ? 'https://api.stg.tidnex.dev/v3' 
        : 'https://api.xendit.co/v3';
  }
} 